python/unblob/file_utils.py

```diff
@@ -355,17 +355,23 @@ def parse(
         return struct_parser(file)
 
 
-def get_endian(file: File, big_endian_magic: int) -> Endian:
+def get_endian(file: File, big_endian_magic: int, endian_len: int = 4) -> Endian:
```

Comment:
I would probably introduce a new `get_endian2` for the 2 byte magic case instead of complicating the existing one.

You can keep it, but please change the signature, as the code will not work with arbitrary `endian_len` - the only supported options are: 2 and 4:

```py
def get_endian(file: File, big_endian_magic: int, endian_len: Literal[2] | Literal[4] = 4) -> Endian:
```

Reply:
@rxpha3l please take care of this

Reply:
Fixed. Also added test cases to `TestGetEndian` to cover both `endian_len` cases.

Reply:
typing hints are wrong

```
/home/runner/work/unblob/unblob/tests/test_file_utils.py:364:62 - error: Argument of type "int" cannot be assigned to parameter "endian_len" of type "Literal[2, 4]" in function "get_endian"
    Type "int" is not assignable to type "Literal[2, 4]"
      "int" is not assignable to type "Literal[2]"
      "int" is not assignable to type "Literal[4]" (reportArgumentType)
  /home/runner/work/unblob/unblob/tests/test_file_utils.py:390:48 - error: Argument of type "int" cannot be assigned to parameter "endian_len" of type "Literal[2, 4]" in function "get_endian"
    Type "int" is not assignable to type "Literal[2, 4]"
      "int" is not assignable to type "Literal[2]"
      "int" is not assignable to type "Literal[4]" (reportArgumentType)
```